goodmap/core_api.py

```diff
@@ -1,22 +1,22 @@
 import importlib.metadata
+import uuid
 
+import deprecation
 from flask import Blueprint, jsonify, make_response, request
 from flask_babel import gettext
 from flask_restx import Api, Resource, fields
 from platzky.config import LanguagesMapping
 
-from goodmap.data_models.location import Location
-
-from .core import get_queried_data
-from .formatter import prepare_pin
+from goodmap.core import get_queried_data
+from goodmap.formatter import prepare_pin
 
 
 def make_tuple_translation(keys_to_translate):
     return [(x, gettext(x)) for x in keys_to_translate]
 
 
 def core_pages(
-    database, languages: LanguagesMapping, notifier_function, csrf_generator
+    database, languages: LanguagesMapping, notifier_function, csrf_generator, location_model
```

Comment:
Is adding a new argument necessary? Now `location_model` is extracted  in `goodmap.py` from `database`, which is passed here as an argument anyway. Maybe we could move that extraction code here, to `core_pages`?

Reply:
code from goodmap.py:
```
   location_model = create_location_model(location_obligatory_fields)

    app.db = goodmap_db_extended_app(app.db, location_model)

    CSRFProtect(app)

    cp = core_pages(
        app.db, languages_dict(config.languages), app.notify, generate_csrf, location_model
    )
```


if we created location_model inside core_pages then we alse would have to move extending db into `core_pages` which would seriously extend what `core_pages` does.

SRP -single responsibility principle. Let it create core_pages. :)

But if you can see other way of doing it, that would be great.

Reply:
You are right, there is something more going on here. We get location_model from app.db, then we use it to extend the app.db and so now app.db is not the same.  Strange, perhaps it could be done differently, but that's for another story.